### PR TITLE
Update technologies list, reorganize docs

### PR DESCRIPTION
**Pull request by haworku**
## Summary
Update our technologies list. Also add a technical design docs folder and move docs in that category inside. 

#### Related issues
https://qmacbis.atlassian.net/browse/MR-2580

### CODE REVIEW

File: docs/Technologies.md [hunk 1]
```diff
@@ -1,61 +1,97 @@
...
+## API handling
+
+### GraphQL 
+
+[GraphQL](https://graphql.org/learn/) is a strictly typed protocol built on top of HTTP that exposes your API as a graph of related entities. It uses a schema to define the API and excels in providing easy retrieval of related entities along with any request (see [Queries](https://graphql.org/learn/queries/)).
+
+### Apollo Server
 
-This is a complicated tool. It handles all the GraphQL operations from the client side, and caches the data locally. The docs are good but it's complicated so building stuff and reading the docs at the same time is the most helpful. There are helpful debugging tools for chrome, pretty much required when getting into the nitty gritty of caching behavior.
+[Apollo Server](https://www.apollographql.com/docs/apollo-server/) is the server for our GraphQL API. GraphQL is a schema based, so Apollo Server, reads the schema, sets up the various GraphQL mutations and queries behind the /graphql endpoint and interacts both with auth and with the database to return proper data to our users.
```
Comment (rswerve) [inline docs/Technologies.md:29]:
```suggestion
[Apollo Server](https://www.apollographql.com/docs/apollo-server/) is the server for our GraphQL API. GraphQL is schema based, so Apollo Server reads the schema, sets up the various GraphQL mutations and queries behind the /graphql endpoint, and interacts both with auth and with the database to return proper data to our users.
```

File: docs/Technologies.md [hunk 2]
```diff
@@ -1,61 +1,97 @@
...
+### New Relic
+
+New Relic is used for endpoint monitoring and to observe Open Telemetry data in deployed environments. Read more in [Monitoring](../README.md#monitoring).
 
-#### pa11y with cypress-audit
+### Open Telemetry
 
-We are running pa11y on every page of our form inside of our Cypress tests using the cypress-audit library. This allows us to sign in and fill out the form using Cypress, and only rely on pa11y to scan those pages. This is also one place in our app where we have implemented a ratchet in CI. The issues present when we started testing with pa11y have been ignored, allowing CI to ensure that we aren't adding new errors in the future. Tickets have been filed to address those existing issues.
+Open Telemetry (OTEL) is a ecosystem of tools for collecting data about an application and its services. It provides a standard way to instrument code and is vendor agnostic. We send our OTEL data to New Relic where it helps us assemble tracing of all of our requests and record errors.
```
Comment (rswerve) [inline docs/Technologies.md:47]:
```suggestion
Open Telemetry (OTEL) is an ecosystem of tools for collecting data about an application and its services. It provides a standard way to instrument code and is vendor agnostic. We send our OTEL data to New Relic where it helps us assemble tracing of all of our requests and record errors.
```

File: docs/Technologies.md [hunk 3]
```diff
@@ -1,61 +1,97 @@
...
 
 [Jaeger](https://www.jaegertracing.io/) is used to observe Open Telemetry data in local dev. Read more in the [Jaeger docs](https://www.jaegertracing.io/docs/1.35/getting-started/) or visit port [16686](http://localhost:16686)) on local dev.
 
-### New Relic
+## React Web Application
 
-New Relic is used for endpoint monitoring and to observe Open Telemetry data in deployed environments. Read more in [Monitoring](../README.md#monitoring).
+### Create React App
+
+[Create React App](https://facebook.github.io/create-react-app) builds and configures our React client, including typescripts, styles, and static assets. See also [react-scripts](https://github.com/facebook/create-react-app/tree/master/packages/react-scripts).
```
Comment (rswerve) [inline docs/Technologies.md:57]: not sure what "typescripts" are

File: docs/Technologies.md [hunk 4]
```diff
@@ -1,61 +1,97 @@
...
+
+[React Router](https://reactrouter.com/) is a React-based tool for declarative routing, allowing us to configure routes with their associated URLs, handle links and navigation within the application, and use data parameters in our urls successfully.
+
+### Formik
+
+[Formik](https://formik.org/docs/overview) is a React-based tool we use for web form development. It holds key information about the current state of a form on the page in a shared context, which makes validations and conditional logic within the form much simpler to maintain.
+
+### United States Web Design System (USWDS)
+
+The [USWDS United States Web Design System](https://designsystem.digital.gov/) is the standard for federal government web design. It used to provide the main scss classes across the application, along with [`@trussworks/react-uswds`](https://github.com/trussworks/react-uswds) as the component library.
```
Comment (rswerve) [inline docs/Technologies.md:69]:
```suggestion
The [USWDS United States Web Design System](https://designsystem.digital.gov/) is the standard for federal government web design. It is used to provide the main scss classes across the application, along with [`@trussworks/react-uswds`](https://github.com/trussworks/react-uswds) as the component library.
```

File: docs/Technologies.md [hunk 5]
```diff
@@ -1,61 +1,97 @@
...
+
+## Testing
+
+More about the MC-Review testing approach can be found in [here](technical-design/testingApproach.md).
+
+### Launch Darkly
+
+[LaunchDarkly](https://launchdarkly.com/implementation/) is third party feature flag management tool that includes a online dashboard for viewing/auditing/updating flags and SDKs that are integrated into our codebase. The use of feature flag with LaunchDarkly allows us de-couple the act of delivering code from the act of enabling a new feature to subsets of users.
```
Comment (rswerve) [inline docs/Technologies.md:77]:
```suggestion
[LaunchDarkly](https://launchdarkly.com/implementation/) is third party feature flag management tool that includes a online dashboard for viewing/auditing/updating flags and SDKs that are integrated into our codebase. The use of feature flags with LaunchDarkly allows us to de-couple the act of delivering code from the act of enabling a new feature to subsets of users.
```

File: docs/Technologies.md [hunk 6]
```diff
@@ -1,61 +1,97 @@
...
+
+[Jest](https://jestjs.io/) is a test runner and framework used for unit tests across the entire codebase.
+
+### Cypress
+
+[Cypress](https://www.cypress.io/) is a test runner and framework for end to end testing in a live browser. This is configured in the main application `/cypress`. Test runs for the application can be seen in Github Actions or in [Cypress Dashboard](https://dashboard.cypress.io) for team developers.
+
+### React Testing Library
+
+[Testing library](https://testing-library.com) provides utility functions for testing DOM elements. We use the [version](https://testing-library.com/docs/react-testing-library/intro) specific to React. React Testing Library facilitates tests that find and interact with elements there like a user would. It also allows provides a useful set of [matchers](https://github.com/testing-library/jest-dom#custom-matchers).
```
Comment (rswerve) [inline docs/Technologies.md:89]:
```suggestion
[Testing library](https://testing-library.com) provides utility functions for testing DOM elements. We use the [version](https://testing-library.com/docs/react-testing-library/intro) specific to React. React Testing Library facilitates tests that find and interact with elements like a user would. It also provides a useful set of [matchers](https://github.com/testing-library/jest-dom#custom-matchers).
```

File: docs/technical-design/testingApproach.md [hunk 7]
```diff
@@ -11,24 +13,20 @@
 
 ## Frontend Component Tests
 
-These tests are written using Jest and Testing Library.
-
-We test all our individual components that are big enough to be in Storybook
-
--   Assert that things render the expected way given different props
+These tests are written using Jest and Testing Library. We test all our individual components that are big enough to be in Storybook and focus on assertions that confirm that elements render as expect given different sets of props and user interactions.
```
Comment (rswerve) [inline docs/technical-design/testingApproach.md:16]:
```suggestion
These tests are written using Jest and Testing Library. We test all our individual components that are big enough to be in Storybook and focus on assertions that confirm that elements render as expected given different sets of props and user interactions.
```

File: docs/technical-design/testingApproach.md [hunk 8]
```diff
@@ -11,24 +13,20 @@
...
+-   Test that the form validations run, given some values, the right error renders
+-   Test that conditional questions display correctly
     -   E.g. if you select “other” is the “other” field displayed?
 -   Test accessibility actions work
     -   E.g. AX focus changes on error
 
 ## Cypress Integration Tests
 
-These tests run against a fully deployed version of the app. Tests that require moving around between different pages of the app, which would require extensive fragile mocking as component tests, should be done here
+These end to end tests are written using Cypress and Cypress Testing Library. They are run against a fully deployed version of the app, specifically for test cases that require moving around between different pages, users, and states of the application. While extensive fragile mocking as component tests, should be done here
```
Comment (rswerve) [inline docs/technical-design/testingApproach.md:29]: don't really understand the second sentence here

File: docs/Technologies.md (hunk 3 above)
Comment (haworku) [inline docs/Technologies.md:57]:
```suggestion
[Create React App](https://facebook.github.io/create-react-app) compiles, builds and configures our React client, including scripts, styles, and static assets. See also [react-scripts](https://github.com/facebook/create-react-app/tree/master/packages/react-scripts).
```

Comment (haworku) [reply docs/Technologies.md:57]: gotcha. reworded a bit below lmk if that is still unclear. wanted to help indicate that this tool is doing the frotnend ts to js compilation as well as bundling, building, configuring the web app

File: docs/technical-design/testingApproach.md (hunk 8 above)
Comment (haworku) [inline docs/technical-design/testingApproach.md:29]:
```suggestion
These end to end tests are written using Cypress and Cypress Testing Library. They are run against a fully deployed version of the app, specifically for test cases that require moving around between different pages, users, and states of the application. In addition, tests that would require extensive mocking as Jest tests should be done with Cypress instead.
```

**Review by mojotalantikite — APPROVED** [review]

**Pull request merged**